src/main/java/codeu/controller/AdminServlet.java

```diff
@@ -1,50 +1,218 @@
+    HashMap<UUID, Integer> userMessageCounts = new HashMap<UUID, Integer>();
+
+    int totalMessageCount = 0; // total number of messages sent by all users
+    int totalWordCount = 0; // total number of words sent by all users
+
+    for (Conversation c: allConversations) {
+
+      messagesInConversation = messageStore.getMessagesInConversation(c.getId());
+      totalMessageCount += messagesInConversation.size();
+
+      for (Message m: messagesInConversation) {
+
+        int previousMessageCount = userMessageCounts.getOrDefault(m.getAuthorId(), 0);
+        userMessageCounts.put(m.getAuthorId(), previousMessageCount + 1);
 
-    @Override
-    public void init() throws ServletException {
-        super.init();
+        totalWordCount += m.getContent().split("\\s+").length;
+      }
 
-        setAdminUsernames();
     }
 
-    void setAdminUsernames() {
-        admins = new HashSet<String>();
-        admins.add("daniel");
-        admins.add("leslie");
-        admins.add("serena");
-        admins.add("shana");
-        admins.add("kyra");
+    // compute most active user
+    Map.Entry<UUID, Integer> maxMessageCountUser = null;
+
+    for (Map.Entry<UUID, Integer> entry: userMessageCounts.entrySet()) {
+
+      if (maxMessageCountUser == null || entry.getValue() > maxMessageCountUser.getValue()) {
+        maxMessageCountUser = entry;
+      }
+
+    }
+    if (maxMessageCountUser != null && userStore.getUser(maxMessageCountUser.getKey()) != null) {
+      mostActiveUser = userStore.getUser(maxMessageCountUser.getKey()).getName();
+    }
+
+    String avgMessagesPerConvo;
+    if (allConversations.size() != 0) {
+      avgMessagesPerConvo = String.format("%.3f", (double) totalMessageCount / allConversations.size());
+    } else {
+      avgMessagesPerConvo = "0.000";
+    }
+
+    String avgWordsPerMessage;
+    if (totalMessageCount != 0) {
+      avgWordsPerMessage = String.format("%.3f", (double) totalWordCount / totalMessageCount);
+    } else {
+      avgWordsPerMessage = "0.000";
+    }
+
+    // Set attributes
+    request.setAttribute("numberOfUsers", allUsers.size());
+    request.setAttribute("newestUser", newestUser);
+    request.setAttribute("mostActiveUser", mostActiveUser);
+
+    request.setAttribute("numberOfConversations", allConversations.size());
+    request.setAttribute("numberOfMessages", totalMessageCount);
+    request.setAttribute("avgMessagesPerConvo", avgMessagesPerConvo);
+    request.setAttribute("avgWordsPerMessage", avgWordsPerMessage);
+
+  }
+
+  void addTestData() {
+    int numberOfUsers = ThreadLocalRandom.current().nextInt(10, 101);
+    int numberOfConversations = ThreadLocalRandom.current().nextInt(5, 21);
+    int numberOfMessages = ThreadLocalRandom.current().nextInt(100, 1001);
+
+    User newUser;
+    for (int i = 0; i < numberOfUsers; i++) {
+      newUser = new User(UUID.randomUUID(), "user"+i, "password",
+        Instant.ofEpochSecond(ThreadLocalRandom.current().nextInt(1, 10001)));
```

Comment:
Micronit: The Google style guide indents four spaces on line continuations like this, so it's clear that it's really just part of the line above it. Up to you if you'd like to change it (or use https://github.com/google/google-java-format).